velox/functions/sparksql/RaiseError.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/Macros.h"
+namespace facebook::velox::functions::sparksql {
+
+template <typename T>
+struct RaiseErrorFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  template <typename TInput>
+  FOLLY_ALWAYS_INLINE void callNullable(
+      UnknownValue& result,
+      const TInput* input) {
+    if (input) {
+      VELOX_FAIL(*input);
+    }
+    VELOX_FAIL("");
```

Comment:
As we are throwing system error, the fuzzer test fails with raise_error function. Do you think it makes sense to throw system error instead of user error here? @PHILO-HE 

Quoted from Masha:
> (1) User error is an error caused by the user of the library; (2) System error indicates a bug in Velox or some failure in the environment.

Reply:
A similar function in Presto is throwing user error: https://github.com/facebookincubator/velox/commit/194772c823e73f3847ab1754a70bcda55a6c0679#diff-2a2739960fe78b23b6ac90af85cef58647341837a9cfec78d3d66eb90ea4b1b0.

Reply:
This exception seems an exceptional one according to Masha's classifying, as this function just throws it.
I note Spark throws `java.lang.RuntimeException`. Maybe, we can use either one.

Reply:
I am both ok , but spark just return runtime error not user error. After @PHILO-HE 's input, I can make the change.

Reply:
> I'm more inclined to throw user error as this error is caused by the user and is not a bug of Velox.
> 
> > spark just return runtime error not user error.
> 
> Gluten catches any std::exception from native and throw a runtime exception to Java through JNI, so I assume both errors should work. How do you think?

oh, I see , good to know,  so let's change to throw a user error instead.

@rui-mo could you share the code of `Gluten catches any std::exception from native and throw a runtime exception to Java through JNI`

Reply:
Gluten defines the macros JNI_METHOD_START and JNI_METHOD_END and wraps each JNI call inside it as below. Would you confirm if it works for raise_error? Thanks.

Links:
https://github.com/apache/incubator-gluten/blob/main/cpp/core/jni/JniError.h#L25-L38
https://github.com/apache/incubator-gluten/blob/main/cpp/core/jni/JniWrapper.cc#L374C3-L374C19